File: primer_trimestre/poo/java/proyectos_java/ejercicio_1_java/src/discount.java
public class discount extends Employee {
    private int riskLevel;

    public discount( int riskLevel) {
        super();
        this.riskLevel = riskLevel;
    }
    public int getRiskLevel() {
        return this.riskLevel;
    }
    public double swARL(){
        return switch (this.riskLevel) {
            case 1 -> 0.00522;
            case 2 -> 0.01044;
            case 3 -> 0.02436;
            case 4 -> 0.04350;
            case 5 -> 0.06960;
            default -> 0;
        };
    }
    public double calculateDiscount(){
        return getBaseSalary() * 0.08 + getBaseSalary() * swARL();
    }
}

File: primer_trimestre/poo/php/ejercicios_php/first/libreria/Discount.php
<?php
class Discount extends Employee {
    private int $riskLevel;

    public function __construct(int $riskLevel) {
        $this->riskLevel = $riskLevel;
    }

    public function getRiskLevel(): int {
        return $this->riskLevel;
    }

    public function swARL(): float {
        return match ($this->riskLevel) {
            1 => 0.00522,
            2 => 0.01044,
            3 => 0.02436,
            4 => 0.04350,
            5 => 0.06960,
            default => 0,
        };
    }

    public function calculateDiscount(): float {
        return $this->getBaseSalary() * 0.08 + $this->getBaseSalary() * $this->swARL();
    }
}
?>
